libs/partners/milvus/langchain_milvus/vectorstores/milvus.py

```diff
@@ -1349,3 +1494,41 @@ def upsert(  # type: ignore
                 "Failed to upsert entities: %s error: %s", self.collection_name, exc
             )
             raise exc
+
+    def _create_ranker(
+        self,
+        ranker_type: Optional[Literal["rrf", "weighted"]],
+        ranker_params: dict,
+    ) -> BaseRanker:
+        """A Ranker factory method"""
+        embeddings_functions: List[EmbeddingType] = (
+            self.embeddings if self._is_hybrid else [self.embeddings]
+        )
+        default_weights = [1.0] * len(embeddings_functions)
+        if not ranker_type:
+            return WeightedRanker(*default_weights)
+
+        if ranker_type == "weighted":
+            weights = ranker_params.get("weights", default_weights)
+            return WeightedRanker(*weights)
+        elif ranker_type == "rrf":
+            k = ranker_params.get("k", None)
+            if k:
+                return RRFRanker(k)
+            return RRFRanker()
+        else:
+            logger.error(
+                "Ranker %s does not exist. "
+                "Please use on of the following rankers: %s, %s",
+                ranker_type,
+                "weighted",
+                "rrf",
+            )
+            raise ValueError("Unrecognized ranker of type %s", ranker_type)
+
+    def _get_vector_fields_as_list(self) -> list[str]:
+        return (
+            [self._vector_field]
+            if not isinstance(self._vector_field, list)
+            else self._vector_field
```

Comment:
```suggestion
            self._vector_field
            if isinstance(self._vector_field, list)
            else [self._vector_field]
```
easier to understand I think